### PR TITLE
Add thumbnail for media messages

### LINKED ISSUE #754: Send Media with Custom Thumbnail
How to send media (image/video) with custom thumbnail.
for example, i send image A, but when downloaded the image changed to B

on WWebJS while sending video just showing black thumbnail,
Screenshot_2021-07-12-05-21-18-66_7352322957d4404136654ef4adb64504

i've tried using extra options, but idk how to make it :D

my code like this

```javascript
                       client.sendMessage(from, MessageMedia.fromFilePath('berli.jpg'), {
                            extra: {
                                type: "image",
                                body: "base64...",
                                quotedMsg: {
                                    "type": "video",
                                    "deprecatedMms3Url": "https://mmg.whatsapp.net/d/f/AnIRJ-om7fiSurdksss5W8pe1sZM80tl15QU171Z6uQv.enc",
                                    "directPath": "/v/t62.7161-24/26676941_191606646250607_998074438285213616_n.enc?ccb=11-4&oh=27eb011945d192938657fab339960590&oe=6110BEA0",
                                    "mimetype": "video/mp4",
                                    "filehash": "ymACSpjzXi3pdGsSe59YePkuWPtoslo3YPwb/CvYCHQ=",
                                    "encFilehash": "wqgZMsXdjCV9ZHPactotvqOqOfQlzvlsv7rHFC4Gw+w=",
                                    "size": 1268322,
                                    "height": 1024,
                                    "width": 576,
                                    "duration": "800",
                                    "mediaKey": "15+THwjZ9QoNouX4sqVDe3tEJD0qPaH9L0/H4gNbBAQ=",
                                    "mediaKeyTimestamp": 1626138615,
                                    "caption": "SLEYER-07",
                                    "body": base64...,
                                    "interactiveAnnotations": [],
                                    "isViewOnce": false
                                }, 
                                quotedParticipant: '0@s.whatsapp.net', 
                                quotedRemoteJid: 'status@broadcast',
                                mediaData: {
                                    "type": "image",
                                    "mediaStage": "RESOLVED",
                                    "size": 390491,
                                    "filehash": "L8jtradfBUC1AfqgpaZnnoog4jLVXR3v/gEAw3anVYE=",
                                    "mimetype": "image/jpeg",
                                    "mediaBlob": null,
                                    "fullHeight": 512,
                                    "fullWidth": 512,
                                    "aspectRatio": 1,
                                    "animationDuration": 0,
                                    "animatedAsNewMsg": false,
                                    "isViewOnce": false,
                                    "preview": {
                                        "_retainCount": 1,
                                        "_inAutoreleasePool": false,
                                        "released": false,
                                        "_b64": "base64...",
                                        "_url": "data:image/jpeg;base64,...",
                                        "_mimetype": "image/jpeg"
                                    },
                                    "_swStreamingSupported": false,
                                    "_listeningToSwSupport": false,
                                    "isVcardOverMmsDocument": false
                                }
                            }
                        })

### PR DESCRIPTION
**Pull request by Darker935**
#55
This is just a base, i accept as many suggestions as possible 

**thumbnail needs to be a MessageMedia instance**

### CODE REVIEW

File: src/util/Injected.js [hunk 1]
```diff
@@ -152,7 +155,6 @@ exports.LoadUtils = () => {
         });
 
         const message = {
-            ...options,
```
Comment (tuyuribr) [inline src/util/Injected.js:155]: I really want the ...options to overwrite all other options. But this can break something, we must be careful

**Review by purpshell — APPROVED** [review]
lgtm

**Review by Gimenz — APPROVED** [review]
lgtm

File: src/util/Injected.js [hunk 2]
```diff
@@ -166,7 +168,8 @@ exports.LoadUtils = () => {
             ...locationOptions,
             ...attOptions,
             ...quotedMsgOptions,
-            ...vcardOptions
+            ...vcardOptions,
+            ...options
```
Comment (pedroslopez) [inline src/util/Injected.js:172]: Why the move of `...options`?

**Comment by pedroslopez** [comment]
Since this is an override for custom thumbnails and isn't a standard whatsapp thing, I'm going to close this in favor of supplying it as part of `extra` options when sending the message. #1103 moved the extra options to be applied after everything else, so you should be able to override anything with it now.

**Pull request closed without merging**